x-pack/platform/plugins/shared/security/server/authentication/providers/saml.test.ts

```diff
@@ -303,7 +338,97 @@ describe('SAMLAuthenticationProvider', () => {
       expect(mockOptions.client.asInternalUser.transport.request).toHaveBeenCalledWith({
         method: 'POST',
         path: '/_security/saml/authenticate',
-        body: { ids: ['some-request-id'], content: 'saml-response-xml', realm: 'test-realm' },
+        body: {
+          ids: [mockSAMLSet1.requestId],
+          content: mockSAMLSet1.samlResponse,
+          realm: 'test-realm',
+        },
+      });
+    });
+
+    describe('Multiple "concurrent" login requests', () => {
```

Comment:
Question: Open to ideas that would actually mimic the multiple concurrent logins; I tried a bunch of things, but I kept breaking other stuff.

So for now, this tests the state mgmt around requestIds